pkg/migrations/sql/versions/dev/0.7.2-dev/2-event_name.sql

```diff
@@ -0,0 +1,114 @@
+/*
+    Remove the event_name_check check constraint on SCHEMA_TRACING.event if it exists
+    You can't drop the constraint if compression is turned on for the table.
+    If compression is on, turn it off.
+
+*/
+DO $block$
+DECLARE
+    _is_timescaledb_installed boolean = false;
+    _is_timescaledb_oss boolean = true;
+    _timescaledb_version_text text;
+    _timescaledb_major_version int;
+    _timescaledb_minor_version int;
+    _is_compression_available boolean = false;
+    _compression_enabled boolean = false;
+BEGIN
+
+    -- determine whether the constraint even exists
+    -- if it doesn't, bail early
+    IF
+    (
+        SELECT count(*) = 0
+        FROM pg_constraint c
+        INNER JOIN pg_class t on (c.conrelid = t.oid)
+        INNER JOIN pg_namespace n on (c.connamespace = n.oid and t.relnamespace = n.oid)
+        WHERE n.nspname = 'SCHEMA_TRACING'
+        AND t.relname = 'event'
+        AND c.conname = 'event_name_check'
+        AND c.contype = 'c'
+    )
+    THEN
+        RETURN;
+    END IF;
+
+    /*
+        These functions do not exist until the
+        idempotent scripts are executed, so we have
+        to deal with it "manually"
+        SCHEMA_CATALOG.get_timescale_major_version()
+        SCHEMA_CATALOG.is_timescaledb_oss()
+        SCHEMA_CATALOG.is_timescaledb_installed()
+        SCHEMA_CATALOG.is_multinode()
+        SCHEMA_CATALOG.get_default_chunk_interval()
+        SCHEMA_CATALOG.get_staggered_chunk_interval(...)
+    */
+    SELECT count(*) > 0
+    INTO STRICT _is_timescaledb_installed
+    FROM pg_extension
+    WHERE extname='timescaledb';
+
+    IF _is_timescaledb_installed THEN
+        SELECT extversion INTO STRICT _timescaledb_version_text
+        FROM pg_catalog.pg_extension
+        WHERE extname='timescaledb'
+        LIMIT 1;
+
+        _timescaledb_major_version = split_part(_timescaledb_version_text, '.', 1)::INT;
+        _timescaledb_minor_version = split_part(_timescaledb_version_text, '.', 2)::INT;
+
+        _is_compression_available = CASE
+            WHEN _timescaledb_major_version >= 2 THEN true
+            WHEN _timescaledb_major_version = 1 and _timescaledb_minor_version >= 5 THEN true
+            ELSE false
+        END;
+
+        IF _timescaledb_major_version >= 2 THEN
+            _is_timescaledb_oss = (current_setting('timescaledb.license') = 'apache');
+        ELSE
+            _is_timescaledb_oss = (SELECT edition = 'apache' FROM timescaledb_information.license);
+        END IF;
+    END IF;
+
+    IF _is_timescaledb_installed AND (NOT _is_timescaledb_oss) AND _is_compression_available THEN
+        -- is compression enabled for the table?
```

Comment:
you need to decompress all chunks here before turning compression off, otherwise disabling compression will fail.

Reply:
This probably means we also should think about how to alter our upgrade tests to catch such bugs

Reply:
This makes me think that we should NOT have an upgrade script for this particular change. If the event table is gigantic, applying this change could be a very long outage. What do you think?

Reply:
The only other way is dropping the table

Reply:
I wanted to suggest that it might be "acceptable" for us to force this onto the users who are using tracing in beta: to tell them to manually run the necessary steps. Unfortunately, because all users have the schema installed, this also impacts users who are on `0.7.0` and do not have `-enable-feature=tracing` set.

Reply:
This has been marked as resolved, but I'm not sure why. @jgpruitt can you add some context?

Reply:
I unresolved this because I added the code to decompress any compressed chunks before disabling compression and dropping the constraints. I also recompress those chunks after dropping the constraint.

Do we want to discuss more?

Reply:
Okay, I think that hits the right middle ground.

Just to be explicit: with this change, we make the constraint dropping fully automatic, with the caveat that any users with an existing promscale tracing installation will have to assess whether uncompressing and recompressing the `event` hypertable will work for them. As you already mentioned, it could cause a long downtime, or even use more disk than is available, and come to a grinding halt.

Do we want to provide users with instructions for how to determine if this will go smoothly? Or do we assume that users who were using the beta feature would know how to do that?

Reply:
It wouldn't hurt to leave something in the release notes. I don't know that there is anything to tell them to DO, just that the upgrade may take a while if the event table is large.

Reply:
Nope. Current consensus is leaning towards starting a CHANGELOG.md file in the repo, but it's not decided, and nothing has been done.

Reply:
> you need to decompress all chunks here before turning compression off, otherwise disabling compression will fail.

@cevian did I satisfy this change request?